freqtrade/exchange/exchange.py

```diff
@@ -1428,14 +1575,38 @@ async def _async_get_trade_history_id(self, pair: str,
             # of up to an hour.
             # e.g. Binance returns the "last 1000" candles within a 1h time interval
             # - so we will miss the first trades.
-            t = await self._async_fetch_trades(pair, since=since)
+            t = await self._async_fetch_trades(pair, since=since, datadir=datadir)
             # DEFAULT_TRADES_COLUMNS: 0 -> timestamp
             # DEFAULT_TRADES_COLUMNS: 1 -> id
             from_id = t[-1][1]
+            logger.debug(
+                "Did not cache the intermediate trades since %s with len=%s and from_id=%s", since,
+                len(t), from_id)
             trades.extend(t[:-1])
         while True:
-            t = await self._async_fetch_trades(pair,
-                                               params={self._trades_pagination_arg: from_id})
+            assert from_id is not None
+
+            t = []
+            success_cache_read = False
+            if datadir:
+                tmpdata_file = self._intermediate_trades_file(datadir, pair, from_id)
+                if os.path.isfile(tmpdata_file):
```

Comment:
Please use `Pathlib` for directory operations, not `os.path`. 

I've worked hard about a year ago to get rid of using os.path, as using `os.path` can be problematic when using different environments (windows, linux).
While this usage is not problematic per se, you can make this clearer by having `tmpdata_file` be a Path object (out of the above method).